horovod/common/message.h

```diff
@@ -130,7 +130,7 @@ class RequestList {
 class Response {
 public:
   enum ResponseType {
-    ALLREDUCE = 0, ALLGATHER = 1, BROADCAST = 2, ERROR = 3
+    ALLREDUCE = 0, ALLGATHER = 1, BROADCAST = 2, JOIN = 4, ERROR = 3
```

Comment:
Can you sort the numbers, i.e. JOIN = 3, ERROR = 4.  There's no need for backward compatibility in this layer as all workers must have the same Horovod version.